ui/lib/components/AutoRefreshButton/AutoRefreshButton.tsx

```diff
@@ -0,0 +1,201 @@
+import React, { useCallback, useEffect, useMemo, useState } from 'react'
+import { DownOutlined, LoadingOutlined, SyncOutlined } from '@ant-design/icons'
+import { Spin, Dropdown, Menu, Space } from 'antd'
+import { useSpring, animated } from 'react-spring'
+import { getValueFormat } from '@baurine/grafana-value-formats'
+import { useTranslation } from 'react-i18next'
+import { addTranslationResource } from '@lib/utils/i18n'
+import { useGetSet } from 'react-use'
+
+interface AutoRefreshButtonProps {
+  autoRefreshSecondsOptions: number[]
+  autoRefreshSeconds: number
+  onAutoRefreshSecondsChange: (v: number) => void
+  onRefresh: () => Promise<void>
+  disabled?: boolean
+}
+
+const translations = {
+  en: {
+    refresh: 'Refresh',
+    auto_refresh: {
+      title: 'Auto Refresh',
+      off: 'Off',
+    },
+  },
+  zh: {
+    refresh: '刷新',
+    auto_refresh: {
+      title: '自动刷新',
+      off: '关闭',
+    },
+  },
+}
+
+for (const key in translations) {
+  addTranslationResource(key, {
+    component: {
+      autoRefreshButton: translations[key],
+    },
+  })
+}
+
+export function AutoRefreshButton({
+  autoRefreshSecondsOptions,
+  autoRefreshSeconds,
+  onAutoRefreshSecondsChange,
+  onRefresh,
+  disabled = false,
+}: AutoRefreshButtonProps) {
+  const { t } = useTranslation()
+  const autoRefreshMenu = useMemo(
+    () => (
+      <Menu
+        onClick={({ key }) =>
+          onAutoRefreshSecondsChange(parseInt(key as string))
+        }
+        selectedKeys={[String(autoRefreshSeconds || 0)]}
+      >
+        <Menu.ItemGroup
+          title={t('component.autoRefreshButton.auto_refresh.title')}
+        >
+          <Menu.Item key="0">
+            {t('component.autoRefreshButton.auto_refresh.off')}
+          </Menu.Item>
+          <Menu.Divider />
+          {autoRefreshSecondsOptions.map((sec) => {
+            return (
+              <Menu.Item key={String(sec)}>
+                {getValueFormat('s')(sec, 0)}
+              </Menu.Item>
+            )
+          })}
+        </Menu.ItemGroup>
+      </Menu>
+    ),
+    [autoRefreshSecondsOptions, onAutoRefreshSecondsChange]
+  )
+
+  const [isLoading, setIsLoading] = useState(false)
+
+  const handleRefresh = useCallback(async () => {
+    if (isLoading) {
+      return
+    }
+    setIsLoading(true)
+    await onRefresh()
+    setIsLoading(false)
+  }, [isLoading, onRefresh])
+
+  // Auto refresh
```

Comment:
getTimer and setTimer is a state. Is it a correct behavior that when timer is changed or finished the component needs a re-render? If this behavior is not needed, then you may want a `useRef` instead.